vendor/ddcloud/resource_vip_pool.go

```diff
@@ -95,11 +106,29 @@ func resourceVIPPoolCreate(data *schema.ResourceData, provider interface{}) erro
 
 	providerState := provider.(*providerState)
 	apiClient := providerState.Client()
+	healthMonitorNames := propertyHelper.GetStringSetItems(resourceKeyVIPPoolHealthMonitorNames)
+	healthMonitorIDs := make([]string, len(healthMonitorNames))
+	if healthMonitorNames != nil {
+		// healthMonitorIDs = []string
+		page := compute.DefaultPaging()
+		page.PageSize = 50
+		healthMonitors, err := apiClient.ListDefaultHealthMonitors(networkDomainID, page)
+		if err != nil {
+			return err
+		}
+		for index, healthMonitorName := range healthMonitorNames {
```

Comment:
Would you consider maybe using a map instead of this doubly-nested loop?

Something like:

```go

func getHealthMonitorIDsByName(networkDomainID, apiClient *compute.Client) (map[string]string, error) {
  healthMonitorIdsByName = make(map[string]string)

  page := compute.DefaultPaging()
  for {
    healthMonitors, err := apiClient.ListDefaultHealthMonitors(networkDomainID, page)
    if err != nil {
      return nil, err
    }
    if healthMonitors.IsEmpty() {
      break
    }

    for _, healthMonitor := range healthMonitors.Items {
      healthMonitorIdsByName[healthMonitor.Name] = healthMonitor.ID
    }

    page.Next()
  }

  return healthMonitorIdsByName, nil
}

// and to use it:

if healthMonitorNames != nil {
  healthMonitorIDs = make([]string)

  healthMonitorIDsByName, err := getHealthMonitorIDsByName(apiClient, networkDomainID)
  if err != nil {
    return nil, err
  }
  for _, healthMonitorName := range healthMonitorNames {
    healthMonitorIDs = append(healthMonitorIDs,
      healthMonitorIDsByName[healthMonitorName]
    )
  }
}
```